### PR TITLE
specify polynomial degree for each correlation order and species

### PR DESCRIPTION
**Pull request by gelzinyte**
Added a `sparseM` polynomial degree measure that allows to specify a different polynomial degree for each correlation order and species. Issue #15.

The default is `type = "sparse"` in `degree_params` and the polynomial degree is specified with `maxdeg` in `basis_params`. So to get "sparse" degree parameters: `degree = degree_params()`

`degree_params` with `type = "sparseM"` has to have `Dd` specified (for example at least `Dd = Dict("default" => 17)`), which is how the polynomial degree is specified, so `maxdeg` will in most instances be set to `1.0`.  Then `degree = degree_params(type = "sparseM", Dd = Dict("default" => 17))`  with `maxdeg = 1.0` is equivalent to the "sparse" case above.

A more extended example: 

```
zH = AtomicNumber(:H)
zC = AtomicNumber(:C)
Dn = Dict( "default" => 1.0 )
Dl = Dict( "default" => 1.5 )
Dd = Dict( "default" => 17,
        1 => 17,
        2 => 15,
        (3, zC) => 12, 
        (3, zH) => 4) 

degree = degree_params(
    type = "sparseM",
    Dd = Dd, 
    Dl = Dl, 
    Dn = Dn)
```

So that we pick an intuitive name - what does the "M" stand for in "SparsePSHDegreeM"?

### CODE REVIEW

**Comment by gelzinyte** [comment]
Just a note - forgot to convert atomic numbers to and from strings, will update

**Comment by cortner** [comment]
do you convert atomic numbers to string containing the chemical symbol? That seems the right approach to me. Though could also just have a `read_dict, write_dict` for `AtomicNumber` that does the conversion automatically.

**Comment by gelzinyte** [comment]
I'll

**Comment by gelzinyte** [comment]
Yes I agree, I'll convert atomic numbers to string containing the chemical symbol. That's also what we do to species involved in the rest of making the basis.

**Comment by gelzinyte** [comment]
@cortner could you please take a look?

**Comment by cortner** [comment]
@gelzinyte ouch, this has been lying around for a while. Is this ready to review and merge?

File: src/read_params.jl [hunk 1]
```diff
@@ -8,6 +8,9 @@ export  fill_defaults!
 
 """ Recursively updates nested dictionaries with default parameters"""
 function fill_defaults!(params::Dict; param_key = "fit_params")
+    # THIS IS THE LINE FROM THE polydeg-per-N BRANCH - HOW SHOULD THIS BE
```
Comment (cortner) [inline src/read_params.jl]: @gelzinyte --- I had to resolve a merge conflict here, but didn't know how to correctly merge. Can you please take a look at this?

Comment (gelzinyte) [reply src/read_params.jl]: everything from both branches should have been kept - I've pushed the correct list

**Pull request merged**